heron/scheduler-core/src/java/com/twitter/heron/scheduler/SubmitterMain.java

```diff
@@ -455,9 +507,20 @@ protected URI uploadPackage(IUploader uploader) throws UploaderException {
   }
 
   protected void callLauncherRunner(Config runtime)
-      throws LauncherException, PackingException {
+      throws LauncherException, PackingException, SubmitDryRunResponse {
     // using launch runner, launch the topology
     LaunchRunner launchRunner = new LaunchRunner(config, runtime);
     launchRunner.call();
   }
+
+  protected String renderDryRunResponse(SubmitDryRunResponse resp) {
+    SubmitDryRunRender render = new SubmitDryRunRender(resp);
+    DryRunFormatType formatType = Context.dryRunFormatType(config);
+    switch (formatType) {
```

Comment:
The main classes don't leverage the `DryRunRender` interface. We'd have more flexibility by not binding the `DryRunRender` impls into the same parent class and instead doing something like this:
```
DryRunRender render = null;
switch (formatType) {
  case RAW : render = new SubmitRawDryRunRenderer();
  case TABLE: render = new SubmitTableDryRunRenderer();
  default:
    throw new IllegalArgumentException(
         String.format("Unexpected rendering format: %s", formatType)); 
}
return render.render();
```

If the two renderers need to share code, they could use inheritance or composition.